modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -133,556 +173,523 @@
-		return
+	var/obj/item/attacking_clothing = attacking_item
 
-	if(istype(attacking_item, /obj/item/glassblowing/metal_cup))
-		var/obj/item/glassblowing/metal_cup/metal_item = attacking_item
-		var/glassblowing_speed = user.mind.get_skill_modifier(/datum/skill/production, SKILL_SPEED_MODIFIER) * DEFAULT_TIMED
-		var/glassblowing_amount = DEFAULT_HEATED / user.mind.get_skill_modifier(/datum/skill/production, SKILL_SPEED_MODIFIER)
+	var/datum/component/reagent_clothing/clothing_component = attacking_clothing.GetComponent(/datum/component/reagent_clothing)
+	if(!clothing_component)
+		fail_message(user, "cannot imbue")
+		return
 
-		if(in_use) //only insert one at a time
-			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
-			return
-		in_use = TRUE
+	if(length(clothing_component.imbued_reagent))
+		fail_message(user, "already imbued")
+		return
 
-		if(forge_temperature < MIN_FORGE_TEMP)
-			fail_message(user, "The temperature is not hot enough to start heating [metal_item]!")
-			return
+	if(!do_after(user, 10 SECONDS, target = src))
+		fail_message(user, "stopped imbuing")
+		return
 
-		if(!metal_item.has_sand)
-			fail_message(user, "There is no sand within [metal_item]!")
-			return
+	for(var/datum/reagent/clothing_reagent in attacking_clothing.reagents.reagent_list)
+		if(clothing_reagent.volume < MINIMUM_IMBUING_REAGENT_AMOUNT)
+			attacking_clothing.reagents.remove_all_type(clothing_reagent.type)
+			continue
 
-		if(!do_after(user, glassblowing_speed, target = src))
-			fail_message(user, "You stop heating up [metal_item]!")
-			return
+		if(is_type_in_typecache(clothing_reagent, disallowed_reagents))
+			balloon_alert(user, "cannot imbue with [clothing_reagent.name]")
+			attacking_clothing.reagents.remove_all_type(clothing_reagent.type)
+			continue
 
-		in_use = FALSE
-		metal_item.has_sand = FALSE
-		metal_item.icon_state = "metal_cup_empty"
-		var/obj/item/glassblowing/molten_glass/spawned_glass = new /obj/item/glassblowing/molten_glass(get_turf(src))
-		user.mind.adjust_experience(/datum/skill/production, 10)
-		COOLDOWN_START(spawned_glass, remaining_heat, glassblowing_amount)
-		return TRUE
+			clothing_component.imbued_reagent += clothing_reagent.type
+			attacking_clothing.name = "[clothing_reagent.name] [attacking_clothing.name]"
 
-	if(istype(attacking_item, /obj/item/food))
-		var/obj/item/food/thing_to_cook = attacking_item
+	attacking_clothing.color = mix_color_from_reagents(attacking_clothing.reagents.reagent_list)
+	balloon_alert_to_viewers("imbued [attacking_clothing]")
+	user.mind.adjust_experience(/datum/skill/smithing, 60)
+	playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+	in_use = FALSE
 
-		if(in_use)
-			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
-			return FALSE
+/// Sets ceramic items from their unusable state into their finished form
+/obj/structure/reagent_forge/proc/handle_ceramics(obj/attacking_item, mob/living/user)
```

Comment:
Rework of how ceramics are made coming after this is merged, I just didn't want to do all of it at once.